src/Controller/Admin/Document/DocumentControllerBase.php

```diff
@@ -170,16 +171,13 @@ protected function addSettingsToDocument(Request $request, Model\Document $docum
     protected function addDataToDocument(Request $request, Model\Document $document): void
     {
         if ($document instanceof Model\Document\PageSnippet) {
-            // if a target group variant get's saved, we have to load all other editables first, otherwise they will get deleted
-
-            if ($request->get('appendEditables')) {
-                // ensure editable are loaded
+            if($request->get('appendEditables') || (interface_exists(TargetingDocumentInterface::class) && $document instanceof TargetingDocumentInterface)) {
```

Comment:
Is if part really required? since the target document requested is served through personalization bundle TargetingPageController or we just keep the else part here?

Reply:
If we skip the section $request->get('appendEditables') in the if part, then the content/document variants for each target group would not be saved